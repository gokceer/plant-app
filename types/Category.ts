export interface CategoryImage {
  id: number;
  name: string;
  alternativeText: string | null;
  caption: string | null;
  width: number;
  height: number;
  formats: any;
  hash: string;
  ext: string;
  mime: string;
  size: number;
  url: string;
  previewUrl: string | null;
  provider: string;
  provider_metadata: any;
  createdAt: string;
  updatedAt: string;
}

export interface Category {
  id: number;
  name: string;
  createdAt: string;
  updatedAt: string;
  publishedAt: string;
  title: string;
  rank: number;
  image: CategoryImage;
}
