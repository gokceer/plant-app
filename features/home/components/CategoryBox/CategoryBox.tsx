import { CategoryImage } from "@/types/Category";
import React from "react";
import {
  Dimensions,
  ImageBackground,
  StyleSheet,
  Text,
  View,
} from "react-native";

const screenWidth = Dimensions.get("window").width;

// horizontal margin -> 24 + 24 and box gap -> 10
const HORIZONTAL_SPACE = 58;
const boxWidth = (screenWidth - HORIZONTAL_SPACE) / 2;

const CategoryBox = ({
  image,
  title,
}: {
  image: CategoryImage;
  title: string;
}) => {
  const boxHeight = boxWidth * 0.9;
  return (
    <View
      style={[styles.categoryBox, { height: boxHeight }]}
      testID="categoryBox"
    >
      <ImageBackground
        source={{ uri: image.url }}
        style={styles.image}
        resizeMode="cover"
      >
        <Text style={styles.categoryTitle} numberOfLines={0}>
          {title}
        </Text>
      </ImageBackground>
    </View>
  );
};

export default CategoryBox;

const styles = StyleSheet.create({
  categoryBox: {
    width: boxWidth,
    marginRight: 10,
    borderRadius: 12,
    overflow: "hidden",
    borderWidth: 0.5,
    borderColor: "#29BB892E",
  },
  image: {
    flex: 1,
  },
  categoryTitle: {
    width: "80%",
    fontSize: 16,
    fontFamily: "RubikMedium",
    color: "#13231B",
    padding: 16,
  },
});
