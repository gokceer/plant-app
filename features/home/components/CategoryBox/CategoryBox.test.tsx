import { CategoryImage } from "@/types/Category";
import { render } from "@testing-library/react-native";
import React from "react";
import CategoryBox from "./CategoryBox";

describe("CategoryBox Component", () => {
  const mockProps = {
    image: {
      url: "https://example.com/image1.jpg",
    } as CategoryImage,
    title: "Test Category",
  };

  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<CategoryBox {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
