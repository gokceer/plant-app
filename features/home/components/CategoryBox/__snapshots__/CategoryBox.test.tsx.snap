// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryBox Component renders correctly and matches snapshot 1`] = `
<View
  style={
    [
      {
        "borderColor": "#29BB892E",
        "borderRadius": 12,
        "borderWidth": 0.5,
        "marginRight": 10,
        "overflow": "hidden",
        "width": 346,
      },
      {
        "height": 311.40000000000003,
      },
    ]
  }
  testID="categoryBox"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "uri": "https://example.com/image1.jpg",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Text
      numberOfLines={0}
      style={
        {
          "color": "#13231B",
          "fontFamily": "RubikMedium",
          "fontSize": 16,
          "padding": 16,
          "width": "80%",
        }
      }
    >
      Test Category
    </Text>
  </View>
</View>
`;
